### PR TITLE
Rename listTables to listRelations to avoid confusion

### PR DESCRIPTION
**Pull request by findepi**
The "tables" is often used to refer to tables only. "Relations"
better indicates the method includes tables, views and materialized
views.

Extracted from https://github.com/trinodb/trino/pull/19832

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/metadata/MetadataListing.java [hunk 1]
```diff
@@ -113,19 +113,19 @@ private static SortedSet<String> doListSchemas(Session session, Metadata metadat
         return ImmutableSortedSet.copyOf(accessControl.filterSchemas(session.toSecurityContext(), catalogName, schemaNames));
     }
 
-    public static Set<SchemaTableName> listTables(Session session, Metadata metadata, AccessControl accessControl, QualifiedTablePrefix prefix)
+    public static Set<SchemaTableName> listRelations(Session session, Metadata metadata, AccessControl accessControl, QualifiedTablePrefix prefix)
```
Comment (mosabua) [inline core/trino-main/src/main/java/io/trino/metadata/MetadataListing.java:116]: At min you should add javadoc here .. but it also looks weird .. you are calling listRelations .. but getting back a set of SchemaTableName objects? Would you not have to rename SchemaTablenName to SchemaRelation as well then to be consistent?

**Review by mosabua — COMMENTED** [review]
Is there no better name for these objects in a schema? Relations is just very abstract. Objects is maybe a bit to broad since other things like functions can be found in a schema ..

What does the SQL spec call them @martint ?

In either way .. I like the idea to rename this .. but it should be consistent and hence also touch the result returned from the methods

**Comment by sajjoseph** [comment]
At my workplace, lot of people refer to tables/views/mv as subjects. listRelations sounds bit of a departure from the nomenclature in rest of the codebase even though I understand the reasoning behind it.

**Comment by losipiuk** [comment]
> > What does the SQL spec call them @martint ?
> 
> See my comment here: [#19832 (comment)](https://github.com/trinodb/trino/pull/19832#discussion_r1399768605)

@martint can you put your thoughts on the replies to your comment on the other PR. The points raised there resonate very much with me.

**Review by raunaqmorarka — APPROVED** [review]

**Comment by mosabua** [comment]
Whoa .. sorry @findepi .. the bot should not have closed a PR .. I will see what happened there .. cc @colebow

**Pull request closed without merging**